align/pnr/write_constraint.py

```diff
@@ -18,6 +18,28 @@ class PnRConstraintWriter:
     def __init__(self):
         pass
 
+    def merge_sametemplate_const(self, all_const, pnr_const):
+        same_template = []
```

Comment:
@kkunal1408 , as merging these constraints is agnostic of the PnR engine, it is preferable to have this logic in the compiler as another compiler step. This also enables making the merge re-usable.

Reply:
This is more like postprocessing of the constraints. I put it here, to accommodate merging in case of auto-generation of constraints is off. In that scenario, can PnR handle multiple user-defined same template constraints?

Reply:
Even without merging the constraints would be reusable.